src/debug_commads.c

```diff
@@ -665,6 +667,32 @@ DEBUG_COMMAND(GCForceBGInvoke) {
   return REDISMODULE_OK;
 }
 
+DEBUG_COMMAND(GCWaitForInvokeAndDrop) {
+  if (argc < 1) {
+    return RedisModule_WrongArity(ctx);
+  }
+  StrongRef ref = IndexSpec_LoadUnsafe(ctx, RedisModule_StringPtrLen(argv[0], NULL), 0);
+  IndexSpec *sp = StrongRef_Get(ref);
+  if (!sp) {
+    return RedisModule_ReplyWithError(ctx, "Unknown index name");
+  }
+  GCTask *task = NULL;
+  uint64_t remaining;
+  while (RedisModule_GetTimerInfo(RSDummyContext, sp->gc->timerID, &remaining, (void**)&task) != REDISMODULE_OK) {
+    // The GC is running, give it a chance to finish and schedule itself again.
+    RedisModule_ThreadSafeContextUnlock(ctx);
```

Comment:
This is supper dangerous, we should never release the GIL on the main thread not even for tests.. I wonder what kind of flakiness issues this can cause. The right thing to do is to block the client and release it when the timer fired.
I wonder if we can make it more general then drop the index after the GC fired, I wonder if there are more stuff that we would like to test on those cases (maybe alter the index? maybe drop and recreate it?). I understand that we can not perform the dropping part from the client because we want atomicity. Maybe we can pass this debug command a Lua script that we will run when the timer fire?

Reply:
Ohh this is waiting for the GC to end... Anyway the comment still hold, we should not release the GIL here but instead block the client and wake it at the proper place...